src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -1757,17 +1757,23 @@ bool WindowImplX11::processEvent(XEvent windowEvent)
         // Key down event
         case KeyPress:
         {
-            // Get the keysym of the key that has been pressed
-            static XComposeStatus keyboard;
-            char buffer[32];
-            KeySym symbol;
-            XLookupString(&windowEvent.xkey, buffer, sizeof(buffer), &symbol, &keyboard);
+            Keyboard::Key key = Keyboard::Unknown;
+
+            // Try each KeySym index (modifier group) until we get a match
+            for (int i = 0; i < 4; ++i)
```

Comment:
There are up to 4 modifier groups per cap, e.g. on [this layout](https://upload.wikimedia.org/wikipedia/commons/2/22/KB_US-International.svg). One for each symbol in the 4 corners of a cap. The API probably doesn't state a maximum value for this because there might be keyboards with even more. :grin: